### PR TITLE
Improved Mod Manager UI

### PR DESCRIPTION
**Pull request by HUSSAR-mtrela**
Improved UI and added new features to the Mod Manager:

- mod sorting
- mod searching
- save/load a list of favorite mods
- compact/expanded view of mod list
- links to mod source/websites
- added optional styling arguments to tooltip functions

Mod Manager in compact view:
image

Mod Manager in expanded view: 
image

Searching for mods with a "black" keyword:
image

Sorting mods by their name:
image

Save Favorite Mods:
image

Load Favorite Mods:
image

Mod Manager show:
-  links to mod info website (if mod_info.lua has url setting)
-  links to github website (if mod_info.lua has github setting)

image

### CODE REVIEW

File: lua/ui/game/tooltip.lua [hunk 1]
```diff
@@ -20,39 +20,42 @@ local Keymapping = import('/lua/keymap/defaultKeyMap.lua').defaultKeyMap
...
 local createThread = false
 
-function CreateMouseoverDisplay(parent, ID, delay, extendedBool, width, forced)
-
-    -- -- values used throughout the function
-
+-- creates a tooltip box from ID table and with optional parameters
+-- @param - ID is a table, e.g. { text = 'tooltip header', body = 'tooltip description' } 
+-- @param - extendedBool indicates whether to just create tooltip header or also tooltip description
+-- @param - width is optional width of tooltip or it is auto calculated based on length of header/description
```
Comment (Garanas) [inline lua/ui/game/tooltip.lua:26]: The format of the new annotations is `--- @param name type description`, as an example:
`---@param width number Optional width of tooltip or it is auto calculated based on length of header/description`.

See also: https://github.com/FAForever/fa/pull/4009/files as an example

File: lua/ui/game/tooltip.lua [hunk 2]
```diff
@@ -193,34 +218,43 @@ function CreateExtendedToolTip(parent, text, desc, width)
...
         local tempTable = false
 
         if desc then
-            tooltip.desc[1] = UIUtil.CreateText(tooltip, "", 12, UIUtil.bodyFont)
+            tooltip.desc[1] = UIUtil.CreateText(tooltip, "", descFontSize, UIUtil.bodyFont)
             tooltip.desc[1].Width:Set(tooltip.Width)
             if text == nil then
-                tooltip.desc[1].Top:Set(tooltip.Top)
-                tooltip.desc[1].Left:Set(tooltip.Left)
+--                tooltip.desc[1].Top:Set(tooltip.Top)
```
Comment (Garanas) [inline lua/ui/game/tooltip.lua:250]: Prefer to remove code instead of commenting it

**Review by Garanas — COMMENTED** [review]
The difference in your screenshots (where the star icon becomes green) doesn't happen:

image

There's some empty space over at the end, we can take that out

image

Great work in general - very exciting!

File: lua/ui/game/tooltip.lua (hunk 1 above)
Comment (HUSSAR-mtrela) [reply lua/ui/game/tooltip.lua:26]: ok, I will correct API comments

**Comment by Garanas** [comment]
Done 👍

**Comment by HUSSAR-mtrela** [comment]
@Garanas I fixed all issues that you found:
- validating domain of URLs before calling OpenURL()
- styling of favorite icon when Random faction is selected
- formatting of API comments in tooltip.lua
- removed dead/commented out code

**Review by Garanas — CHANGES_REQUESTED** [review]
image

The list count doesn't adjust when you toggle filters on / off

image

When you toggle on the favorite mods the list resets, and you end up at the top again. That feels odd

The default color of the stars should be darker, to have a significant difference when you've chosen a random faction. I tried `ff434343`, that works good

@HUSSAR-mtrela Do you think you can find the time to do this in the next few days? I'd like to include this with the release, but for that I'd need to merge it sooner than later 😄

**Comment by HUSSAR-mtrela** [comment]
@Garanas alright, I fixed scroll bar when mods are filtered and I used your color for unchecked favorite toggle. Also, I fixed counter for blacklisted mods and mods missing dependency. 

However, I did not change resting of mod list to top after toggling on the favorite mods because the same happens when filtering mods since both filtering mods and loading favorite mods are using the same function for refreshing mods list. Changing this behavior would be risky and it would add additional complexity. Ideally, this change would be much easier done when I replace (in another PR/release) the custom scroll control used in the Mod Manager with existing Grid control which has built-in scrolling.

**Comment by Garanas** [comment]
Sounds good - and it is looking good. Great work!

**Comment by 4z0t** [comment]
Really like new look, however:
1. I'd really appreciate if it is done in new style, so, there is a class that handles it, lines  has class too and so on
2. Use Layouter instead of LayoutHelpers
3. Star's background doesnt match with ingame one 
image

File: lua/ui/lobby/ModsManager.lua [hunk 3]
```diff
@@ -127,18 +162,19 @@ local modsTags = {
 function CreateDialog(parent, isHost, availableMods, saveBehaviour)
```
Comment (4z0t) [inline lua/ui/lobby/ModsManager.lua:162]: must be a Class

File: lua/ui/lobby/ModsManager.lua [hunk 4]
```diff
@@ -155,31 +191,109 @@ function CreateDialog(parent, isHost, availableMods, saveBehaviour)
     LayoutHelpers.AtHorizontalCenterIn(subtitle, dialogContent, 0)
     LayoutHelpers.AtTopIn(subtitle, dialogContent, 26)
 
+    CreateSortComboBox()
+
+    -- creating a toggle for activating favorite mods
+    local favoriteModsToggle = CreateFavoriteButton(dialogContent, 25, false, true)
+    LayoutHelpers.AtLeftIn(favoriteModsToggle.Icon, dialogContent, 18)
```
Comment (4z0t) [inline lua/ui/lobby/ModsManager.lua:198]: forget LayoutHelpers, use Layouter

Comment (Garanas) [reply lua/ui/lobby/ModsManager.lua:198]: This PR builds on legacy code, it is not a reimplementation. For now, this is fine

**Pull request merged**